api/client.go

```diff
@@ -1143,12 +1172,18 @@ func (c *Client) rawRequestWithContext(ctx context.Context, r *Request) (*Respon
 	checkRetry := c.config.CheckRetry
 	backoff := c.config.Backoff
 	httpClient := c.config.HttpClient
+	headers := c.headers
 	outputCurlString := c.config.OutputCurlString
 	logger := c.config.Logger
 	c.config.modifyLock.RUnlock()
 
 	c.modifyLock.RUnlock()
 
+	// explicitly set the namespace header to the current client
+	if headers != nil {
+		r.Headers.Set(consts.NamespaceHeaderName, headers.Get(consts.NamespaceHeaderName))
```

Comment:
Should we check that `NamespaceHeaderName` key exists. I'm not sure if inserting an empy string header is equivalent to no header.

Reply:
Yeah I think that's good move.